### PR TITLE
MODORDERS-223: Use Json API for loading module schemas

### PR DESCRIPTION
**Pull request by javalkar-varun**
## Purpose

https://issues.folio.org/browse/MODORDERS-223

## Approach

- Enabling /_/jsonSchemas API which provides schemas for acq-models/mod-orders and mod-orders-storage schemas
- This change will be tested with API test once this PR is merged

## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [x] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -260,6 +260,10 @@
                   <key>raml_files</key>
                   <value>${ramlfiles_path}</value>
                 </systemProperty>
+                <systemProperty>
+                  <key>schema_paths</key>
+                  <value>${jsonschema_paths}</value>
```
Comment (piotr-kalashuk) [inline pom.xml]: Looks like definition of `jsonschema_paths` property is missed.

Comment (javalkar-varun) [reply pom.xml]: Added the property.

**Comment by piotr-kalashuk** [comment]
I've just deployed this branch locally and checked how it works
1. `GET /_/jsonSchemas` request returns following
   image
2. Then I tried `GET /_/jsonSchemas?path=acq-models/mod-orders/schemas/composite_po_line.json`
   image
   As can be seen all `$ref` are resolved providing full URL on how to get the content. I am using vagrant box so the okapi URL is http://10.0.2.15:9130
3. If I try to get this resource as instructed, I cannot do this
   image

Probably just to simplify usage of this endpoint, we can define also some of `mod-orders-storage` and `raml-util` (at least `metadata.schema`) schemas?

**Comment by javalkar-varun** [comment]
I have updated the property to add metadata schemas.

**Comment by piotr-kalashuk** [comment]
As far as I can see in [JSON Schemas API](https://github.com/folio-org/raml-module-builder#json-schemas-api):
> Add schema_paths system property to ... specify comma-separated list of directories that should be searched for schema files.
I believe this is a reason why `metadata.schema` is not included
image

File: pom.xml [hunk 2]
```diff
@@ -43,12 +43,13 @@
     <aspectj.version>1.8.13</aspectj.version>
     <module_name>mod-orders</module_name>
     <http.port>8081</http.port>
+    <jsonschema_paths>acq-models/mod-orders/schemas,acq-models/mod-orders-storage/schemas,raml-util/schemas</jsonschema_paths>
```
Comment (pautau-s) [inline pom.xml]: Looks like `acq-models/common/schemas` is missing here.

Comment (piotr-kalashuk) [reply pom.xml]: Just to speed up process I committed the change `acq-models/common/schemas`

image

**Pull request merged**